### PR TITLE
TD-2308 Add support to show helper icon and tooltip

### PR DESCRIPTION
**Pull request by m4manjesh**
Contributes to [TD-2308](https://sce.myjetbrains.com/youtrack/issue/TD-2308/Update-Autocomplete-component-to-support-help-tooltip)

This is the first PR for this ticket to update the Autocomplete component to add support to show a helper icon for the tooltip.

## Changes

- `KeyValueOption` type updated to support optional `tooltip` field.
- New `OptionWithTooltip` component to show helper icon with tooltip 
- New "Key Value Options with Tooltip" story added
- New tests added to check the helper icon and the tooltip is shown if the object contain tooltip fields
-  Updated the `isOptionEqualToValue` check logic to avoid warning in key-value options. 

## Dependencies

- This PR needs to wait until this gets merged #769 

## UI/UX
Figma: https://www.figma.com/proto/s2NuBwBayu8144EGjBCi5d/VIRTO.DATA?node-id=3478-40400&scaling=min-zoom&page-id=281%3A22633&starting-point-node-id=4367%3A27791

Screenshot (262)

Already demoed to @Sowbhagya-ipg 

## Testing notes

- New "Key Value Options with Tooltip" should work as in Figma
- All other stories should work fine as before and should not show helper icon

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [x] ~Branch has been run in docker.~
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

**Comment by Sowbhagya-ipg** [comment]
Checked it. LGTM

File: src/Autocomplete/Autocomplete.tsx [hunk 1]
```diff
@@ -53,7 +55,7 @@ export default function Autocomplete<
           variant={variant}
         />
       )}
-      renderOption={multiple ? Option : undefined}
+      renderOption={multiple ? Option : OptionWithTooltip}
```
Comment (mattcorner) [inline src/Autocomplete/Autocomplete.tsx:58]: Can I not render a tooltip in multiselect mode? Seems like we've coded ourselves out of this option.

Why can't we update the Option component to support mutiple = true | false, option = string | Option with and without tooltip.

Seems like the [renderOption](https://mui.com/material-ui/api/autocomplete/#Autocomplete-prop-renderOption) function interface gives us all the information we'd need to do this.

File: src/Autocomplete/Autocomplete.tsx [hunk 2]
```diff
@@ -98,3 +100,38 @@ function Option(
...
+  props: React.HTMLAttributes<HTMLLIElement>,
+  option: KeyValueOption | string
+) {
+  if (typeof option === "object") {
+    return (
+      <Box
+        {...props}
+        component="li"
+        sx={{
+          justifyContent: "space-between !important"
```
Comment (mattcorner) [inline src/Autocomplete/Autocomplete.tsx:115]: Whats the need for the important here? What are we have to force override?

File: src/Autocomplete/Autocomplete.stories.tsx [hunk 3]
```diff
@@ -112,10 +113,10 @@ export const KeyValueOptions: StoryObj<typeof Autocomplete> = {
...
     multiple: false,
     options: [
-      { key: 1, value: "Option 1" },
-      { key: 2, value: "Option 2" },
-      { key: 3, value: "Option 3" },
-      { key: 4, value: "Option 4" }
+      { key: 1, value: "Option1" },
+      { key: 2, value: "Option2" },
+      { key: 3, value: "Option3" },
+      { key: 4, value: "Option4" }
```
Comment (mattcorner) [inline src/Autocomplete/Autocomplete.stories.tsx:119]: We've lost the space in the text here. Any reason?

**Review by mattcorner — CHANGES_REQUESTED** [review]

File: src/Autocomplete/Autocomplete.stories.tsx (hunk 3 above)
Comment (m4manjesh) [reply src/Autocomplete/Autocomplete.stories.tsx:119]: Copy pasted the wrong object.

File: src/Autocomplete/Autocomplete.tsx (hunk 2 above)
Comment (m4manjesh) [reply src/Autocomplete/Autocomplete.tsx:115]: ScreenShot Tool -20231115091217
 I am forcing style override to achieve this. By default, the li is "justify-content: "flex-start", this was not working by just adding "justifyContent": "space-between", so I am forcing it.

**Review by mattcorner — APPROVED** [review]

**Pull request merged**